src/main/kotlin/com/lovelysystems/gradle/LSAws.kt

```diff
@@ -0,0 +1,114 @@
+package com.lovelysystems.gradle
+
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.getValue
+import org.gradle.kotlin.dsl.invoke
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.kotlin.dsl.registering
+import java.io.ByteArrayOutputStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+
+private const val AWS_GROUP = "aws"
+
+fun Project.awsProject(profile: String, ssoSessionSettings: SsoSessionSettings) {
+
+    tasks {
+
+        /**
+         * Task to setup the AWS SSO session config if not already present. This is required to use AWS SSO and must be
+         * setup only once.
+         */
+        val ssoSessionSetup by tasks.registering {
+            group = AWS_GROUP
+            description = "Setup lovely-sso session config"
+
+            doLast {
+                val userHome = System.getProperty("user.home")
+                val configFile = Paths.get("$userHome/.aws/config")
+                val config = Files.readAllLines(configFile)
+
+                if (!config.contains("[sso-session ${SsoSessionSettings.NAME}]")) {
+                    val sessionConfig = ssoSessionSettings.toAwsConfig()
+                    Files.write(configFile, sessionConfig.toByteArray(), StandardOpenOption.APPEND)
+                    println("Added ${SsoSessionSettings.NAME} session config to $configFile")
+                }
+            }
+        }
+
+        fun runCmd(cmd: String, onError: (AwsError) -> Unit = { error(it.msg) }) {
+            ByteArrayOutputStream().use { errorOut ->
+                val res = project.exec {
+                    errorOutput = errorOut
+                    isIgnoreExitValue = true
+                    commandLine("sh", "-c", cmd)
+                }
+                if (res.exitValue != 0) {
+                    val awsError = AwsError(errorOut.toString().trim())
+                    onError(awsError)
+                }
+            }
+        }
+
+        /**
+         * Task to get AWS SSO credentials for the configured [profile].
+         *
+         * - If the profile is not configured, it will abort and ask to configure the profile first.
+         * - If the profile is configured but the SSO token is expired, it will login to AWS SSO (automatically opens
+         *   browser).
+         */
+        val ssoCredentials by registering {
+            group = AWS_GROUP
+            description = "Get credentials using AWS SSO"
+            dependsOn(ssoSessionSetup)
+
+            doLast {
+                runCmd("[[ \$(aws configure --profile $profile list) && \$? -eq 0 ]]") { error ->
+                    if (error.profileNotFoundError) {
+                        val msg = """
+                        AWS profile '$profile' is not configured:
```

Comment:
As discussed, not the best solution yet but for me it's ok to have it like this.